diff -urN apache_1.3.23/src/main/http_core.c apache_1.3.23_no404/src/main/http_core.c
--- apache_1.3.23/src/main/http_core.c	Thu Jan 17 06:34:32 2002
+++ apache_1.3.23_no404/src/main/http_core.c	Sun Mar 10 14:22:41 2002
@@ -3797,7 +3797,7 @@
     }
 
     if (r->finfo.st_mode == 0 || (r->path_info && *r->path_info)) {
-	ap_log_rerror(APLOG_MARK, APLOG_ERR|APLOG_NOERRNO, r,
+	ap_log_rerror(APLOG_MARK, APLOG_INFO|APLOG_NOERRNO, r,
 		      "File does not exist: %s",r->path_info ?
 		      ap_pstrcat(r->pool, r->filename, r->path_info, NULL)
 		      : r->filename);
