--- main.c.orig	2004-05-05 05:19:51.000000000 +0900
+++ main.c	2004-05-05 05:20:47.000000000 +0900
@@ -4834,7 +4834,7 @@
 chkURLBuffer(Buffer *buf)
 {
     static char *url_like_pat[] = {
-	"https?://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./?=~_\\&+@#,\\$;]*[a-zA-Z0-9_/=\\-]",
+	"h?ttps?://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./?=~_\\&+@#,\\$;]*[a-zA-Z0-9_/=\\-]",
 	"file:/[a-zA-Z0-9:%\\-\\./=_\\+@#,\\$;]*",
 #ifdef USE_GOPHER
 	"gopher://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./_]*",
@@ -4848,7 +4848,7 @@
 	"mailto:[^<> 	][^<> 	]*@[a-zA-Z0-9][a-zA-Z0-9\\-\\._]*[a-zA-Z0-9]",
 #endif
 #ifdef INET6
-	"https?://[a-zA-Z0-9:%\\-\\./_@]*\\[[a-fA-F0-9:][a-fA-F0-9:\\.]*\\][a-zA-Z0-9:%\\-\\./?=~_\\&+@#,\\$;]*",
+	"h?ttps?://[a-zA-Z0-9:%\\-\\./_@]*\\[[a-fA-F0-9:][a-fA-F0-9:\\.]*\\][a-zA-Z0-9:%\\-\\./?=~_\\&+@#,\\$;]*",
 	"ftp://[a-zA-Z0-9:%\\-\\./_@]*\\[[a-fA-F0-9:][a-fA-F0-9:\\.]*\\][a-zA-Z0-9:%\\-\\./=_+@#,\\$]*",
 #endif				/* INET6 */
 	NULL
--- url.c.orig	2004-05-05 05:21:02.000000000 +0900
+++ url.c	2004-05-05 05:23:25.000000000 +0900
@@ -69,6 +69,7 @@
 
 struct cmdtable schemetable[] = {
     {"http", SCM_HTTP},
+    {"ttp", SCM_HTTP},
     {"gopher", SCM_GOPHER},
     {"ftp", SCM_FTP},
     {"local", SCM_LOCAL},
